service/src/test/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityUserServiceTest.java

```diff
@@ -28,88 +30,76 @@
 
     private static final String AUTH_TOKEN = "Bearer token";
     private static final String SERVICE_AUTH_TOKEN = "Bearer service token";
-    private static final String JURISDICTION = "PUBLICLAW";
-    private static final String CASE_TYPE = "CARE_SUPERVISION_EPO";
     private static final String CASE_ID = "1";
-    private static final String CREATOR_USER_ID = "1";
+    private static final String[] USER_IDS = {"1", "2", "3"};
     private static final String LOCAL_AUTHORITY = "example";
+    private static final Set<String> caseRoles = Set.of("[LASOLICITOR]","[CREATOR]");
 
     @Mock
     private AuthTokenGenerator authTokenGenerator;
+
     @Mock
-    private CaseAccessApi caseAccessApi;
+    private CaseUserApi caseUserApi;
     @Mock
     private LocalAuthorityUserLookupConfiguration localAuthorityUserLookupConfiguration;
 
+    @Mock
+    private IdamClient client;
+
     @InjectMocks
     private LocalAuthorityUserService localAuthorityUserService;
 
-    @Test
-    void shouldThrowCustomExceptionWhenValidLocalAuthorityHasNoUsers() throws IllegalArgumentException {
-        given(localAuthorityUserLookupConfiguration.getUserIds(LOCAL_AUTHORITY)).willReturn(
-            ImmutableList.<String>builder().build()
-        );
-
-        assertThatThrownBy(() ->
-            localAuthorityUserService.grantUserAccess(AUTH_TOKEN, CREATOR_USER_ID, CASE_ID, LOCAL_AUTHORITY))
-            .isInstanceOf(NoAssociatedUsersException.class)
-            .hasMessage("No users found for the local authority 'example'");
-    }
+    @BeforeEach
```

Comment:
we seem to have dropped a couple of tests in this file. What happens when invalid local authority? What happens if user already has the role? These circumstances might never be able to happen, but let's test just to make sure it's not going to fall over? What do you reckon? If you're convinced they'll never happen we can leave it

Reply:
i've added a test for invalid local authority now, i have not added one for if the user already has the role as nothing breaks in this instance, it simply updates their role to have it again if they do - same functionality as was there previously

Reply:
let's write the test then. Then whenever anyone asks the question, you can point to the test :)

Reply:
done :D